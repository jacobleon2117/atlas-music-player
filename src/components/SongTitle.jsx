export default function SongTitle() {
  return (
    <>
      <h1 className="text-2xl font-bold">Painted in Blue</h1>
      <p className="text-custom-slate text-base font-normal">Soul Canvas</p>
    </>
  );
}
