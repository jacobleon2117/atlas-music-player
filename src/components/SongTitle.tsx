interface SongTitleProps {
  title: string;
  artist: string;
}

export default function SongTitle({ title, artist }: SongTitleProps) {
  if (!title || !artist) {
    return (
      <div className="animate-pulse">
        <div className="mb-2 h-8 w-3/4 rounded bg-gray-300 dark:bg-gray-700"></div>
        <div className="h-6 w-1/2 rounded bg-gray-300 dark:bg-gray-700"></div>
      </div>
    );
  }

  return (
    <>
      <h1 className="truncate text-2xl font-bold text-primary-dark dark:text-white">
        {title}
      </h1>
      <p className="truncate text-base font-normal text-primary-dark/80 dark:text-primary-sage">
        {artist}
      </p>
    </>
  );
}