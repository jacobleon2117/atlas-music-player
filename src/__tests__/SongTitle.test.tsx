import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

describe('SongTitle', () => {
  it('renders with title and artist', () => {
    const { container } = render(
      <SongTitle
        title="Test Song"
        artist="Test Artist"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders loading state when props are empty', () => {
    const { container } = render(
      <SongTitle
        title=""
        artist=""
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders loading state when props are undefined', () => {
    const { container } = render(
      <SongTitle
        title={undefined}
        artist={undefined}
      />
    );
    expect(container).toMatchSnapshot();
  });
});