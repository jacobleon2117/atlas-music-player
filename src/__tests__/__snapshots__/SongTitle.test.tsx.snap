// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders loading state when props are empty 1`] = `
<div>
  <div
    class="animate-pulse"
  >
    <div
      class="mb-2 h-8 w-3/4 rounded bg-gray-300 dark:bg-gray-700"
    />
    <div
      class="h-6 w-1/2 rounded bg-gray-300 dark:bg-gray-700"
    />
  </div>
</div>
`;

exports[`SongTitle > renders loading state when props are undefined 1`] = `
<div>
  <div
    class="animate-pulse"
  >
    <div
      class="mb-2 h-8 w-3/4 rounded bg-gray-300 dark:bg-gray-700"
    />
    <div
      class="h-6 w-1/2 rounded bg-gray-300 dark:bg-gray-700"
    />
  </div>
</div>
`;

exports[`SongTitle > renders with title and artist 1`] = `
<div>
  <h1
    class="truncate text-2xl font-bold text-primary-dark dark:text-white"
  >
    Test Song
  </h1>
  <p
    class="truncate text-base font-normal text-primary-dark/80 dark:text-primary-sage"
  >
    Test Artist
  </p>
</div>
`;
